typings/algoliasearch.d.ts

```diff
@@ -0,0 +1,1252 @@
+     * @param object
+     * @param cb(err, res)
+     * https://github.com/algolia/algoliasearch-client-js#update-objects---saveobjects
+     */
+    saveObject(object:Object, cb:Function);
+    /**
+     * Add or replace several objects
+     * @param objects
+     * @param cb(err, res)
+     * https://github.com/algolia/algoliasearch-client-js#update-objects---saveobjects
+     */
+    saveObjects(objects:Object[], cb:Function);
+    /**
+     * Update parameters of a specific object
+     * @param object
+     * @param cb(err, res)
+     * https://github.com/algolia/algoliasearch-client-js#update-objects---saveobjects
+     */
+    partialUpdateObject(object:Object, cb:Function);
+    /**
+     * Update parameters of a list of objects
+     * @param objects
+     * @param cb(err, res)
+     * https://github.com/algolia/algoliasearch-client-js#update-objects---saveobjects
+     */
+    partialUpdateObjects(objects:Object[], cb:Function);
+    /**
+     * Delete a specific object
+     * @param objectID
+     * @param cb(err, res)
+     * https://github.com/algolia/algoliasearch-client-js#delete-objects---deleteobjects
+     */
+    deleteObject(objectID:string, cb:Function);
+    /**
+     * Delete a list of objects
+     * @param objectIDs
+     * @param cb(err, res)
+     * https://github.com/algolia/algoliasearch-client-js#delete-objects---deleteobjects
+     */
+    deleteObjects(objectIDs:string[], cb:Function);
+    /**
+     * Delete objects that matches the query
+     * @param query
+     * @param cb(err, res)
+     * https://github.com/algolia/algoliasearch-client-js#delete-by-query---deletebyquery
+     */
+    deleteByQuery(query:string, cb:Function);
+    /**
+     * Delete objects that matches the query
+     * @param query
+     * @param params of the object
+     * @param cb(err, res)
+     * https://github.com/algolia/algoliasearch-client-js#delete-by-query---deletebyquery
+     */
+    deleteByQuery(query:string, params:Object, cb:Function);
+    /**
+     * Wait for an indexing task to be compete
+     * @param taskID
+     * @param cb(err, res)
+     * https://github.com/algolia/algoliasearch-client-js#wait-for-operations---waittask
+     */
+    waitTask(taskID:number, cb:Function);
+    /**
+     * Get an index settings
+     * @param cb(err, res)
+     * https://github.com/algolia/algoliasearch-client-js#get-settings---getsettings
+     */
+    getSettings(cb:Function);
+    /**
+     * Set an index settings
+     * @param settings
+     * @param cb(err, res)
+     * https://github.com/algolia/algoliasearch-client-js#set-settings---setsettings
+     */
+    setSettings(settings:IAlgoliaIndexSettings, cb:Function);
+    /**
+     * Search in an index
+     * @param query string
+     * @param params query parameters
+     * return Promise
+     * https://github.com/algolia/algoliasearch-client-js#search-in-an-index---search
+     */
+    search(query:string, params:IAlgoliaQueryParameters):Promise<IAlgoliaResponse>;
```

Comment:
I think for search we should only advise to use the form:

```js
search({params}, [cb]}
```

query, params is not deprecated but there's no real advantage of using it. Makes the API simpler